pkg/ddl/notifier/events.go

```diff
@@ -426,9 +426,28 @@ func NewFlashbackClusterEvent() *SchemaChangeEvent {
 	}
 }
 
+// NewDropSchemaEvent creates a schema change event whose type is ActionDropSchema.
+func NewDropSchemaEvent(dbInfo *model.DBInfo, tables []*model.TableInfo) *SchemaChangeEvent {
+	return &SchemaChangeEvent{
+		inner: &jsonSchemaChangeEvent{
+			Tp:     model.ActionDropSchema,
+			DBInfo: dbInfo,
+			Tables: tables,
+		},
+	}
+}
+
+// GetDropSchemaInfo returns the database info and tables of the SchemaChangeEvent whose type is ActionDropSchema.
+func (s *SchemaChangeEvent) GetDropSchemaInfo() (dbInfo *model.DBInfo, tables []*model.TableInfo) {
+	intest.Assert(s.inner.Tp == model.ActionDropSchema)
+	return s.inner.DBInfo, s.inner.Tables
+}
+
 // jsonSchemaChangeEvent is used by SchemaChangeEvent when needed to (un)marshal data,
 // we want to hide the details to subscribers, so SchemaChangeEvent contain this struct.
 type jsonSchemaChangeEvent struct {
+	DBInfo          *model.DBInfo        `json:"db_info,omitempty"`
+	Tables          []*model.TableInfo   `json:"tables,omitempty"`
```

Comment:
this event can be very large. Can we only save needed information?

Reply:
The partitioned information is needed. Maybe we should consider only store ids?

Reply:
Updated.